tests/scenario/test_machine_charm/test_cos_agent_e2e.py

```diff
@@ -77,7 +77,16 @@ def vroot(placeholder_cfg_path):
         os.chdir(old_cwd)
 
 
-def test_cos_agent_e2e(vroot):
+@pytest.fixture
+def snap_is_installed(autouse=True):
+    with patch(
+        "machine_charm.GrafanaAgentMachineCharm._is_installed", new_callable=PropertyMock
+    ) as mock_foo:
+        mock_foo.return_value = True
+        yield
+
+
+def test_cos_agent_e2e(vroot, snap_is_installed):
```

Comment:
**change requested** this test is huge and tests too many things. 
We should split every step out to a separate test.